exercises/affine-cipher/affine_cipher_test.rb

```diff
@@ -1,122 +1,149 @@
 require 'minitest/autorun'
 require_relative 'affine_cipher'
 
-class AffineTest < Minitest::Test
+# Common test data version: 2.0.0 8026923
+class AffineCipherTest < Minitest::Test
   def test_encode_yes
-    #skip
-    cipher = Affine.new
-    cipher.addkey(5, 7)
-    assert_equal 'xbt', cipher.encode('yes')
+    # skip
+    cipher = Affine.new(5, 7)
+    plaintext = 'yes'
+    ciphertext = 'xbt'
+    assert_equal ciphertext, cipher.encode(plaintext)
   end
 
   def test_encode_no
     skip
-    cipher = Affine.new
-    cipher.addkey(15, 18)
-    assert_equal 'fu', cipher.encode('no')
+    cipher = Affine.new(15, 18)
+    plaintext = 'no'
+    ciphertext = 'fu'
+    assert_equal ciphertext, cipher.encode(plaintext)
   end
 
-  def test_encode_OMG
+  def test_encode_omg
     skip
-    cipher = Affine.new
-    cipher.addkey(21, 3)
-    assert_equal 'lvz', cipher.encode('OMG')
+    cipher = Affine.new(21, 3)
+    plaintext = 'OMG'
+    ciphertext = 'lvz'
+    assert_equal ciphertext, cipher.encode(plaintext)
   end
 
-  def test_encode_O_M_G
+  def test_encode_o_m_g
     skip
-    cipher = Affine.new
-    cipher.addkey(25, 47)
-    assert_equal 'hjp', cipher.encode('O M G')
+    cipher = Affine.new(25, 47)
+    plaintext = 'O M G'
+    ciphertext = 'hjp'
+    assert_equal ciphertext, cipher.encode(plaintext)
   end
 
   def test_encode_mindblowingly
     skip
-    cipher = Affine.new
-    cipher.addkey(11, 15)
-    assert_equal 'rzcwa gnxzc dgt', cipher.encode('mindblowingly')
+    cipher = Affine.new(11, 15)
+    plaintext = 'mindblowingly'
+    ciphertext = 'rzcwa gnxzc dgt'
+    assert_equal ciphertext, cipher.encode(plaintext)
   end
 
   def test_encode_numbers
     skip
-    cipher= Affine.new
-    cipher.addkey(3, 4)
-    assert_equal 'jqgjc rw123 jqgjc rw', cipher.encode('Testing,1 2 3, testing')
+    cipher = Affine.new(3, 4)
+    plaintext = 'Testing,1 2 3, testing.'
+    ciphertext = 'jqgjc rw123 jqgjc rw'
+    assert_equal ciphertext, cipher.encode(plaintext)
   end
 
   def test_encode_deep_thought
     skip
-    cipher = Affine.new
-    cipher.addkey(5, 17)
-    assert_equal 'iynia fdqfb ifje', cipher.encode('Truth is fiction.')
+    cipher = Affine.new(5, 17)
+    plaintext = 'Truth is fiction.'
+    ciphertext = 'iynia fdqfb ifje'
+    assert_equal ciphertext, cipher.encode(plaintext)
   end
 
   def test_encode_all_the_letters
     skip
-    cipher = Affine.new
-    cipher.addkey(17, 33)
+    cipher = Affine.new(17, 33)
     plaintext = 'The quick brown fox jumps over the lazy dog.'
     ciphertext = 'swxtj npvyk lruol iejdc blaxk swxmh qzglf'
     assert_equal ciphertext, cipher.encode(plaintext)
   end
 
   def test_encode_with_a_not_coprime_to_m
     skip
-    cipher = Affine.new
-    assert_raises(ArgumentError) { cipher.addkey(6, 17) }
+    assert_raises(ArgumentError) { Affine.new(6, 17) }
```

Comment:
What happens if you pass in valid numbers in the wrong order?
eg: `Affine.new(17, 5)`

(This is probably not anything we need to care about here, I'm just wondering.)

Reply:
Well, key_b can essentially be any number whereas key_a must be coprime to 26.

Since it is technically fine for key_b to be coprime to 26 as well (just not required), I think there could be too many pitfalls to making the object determine which value should be key_a.

Reply:
In short 17, 5 is a valid key.  5, 17 is also a valid key, but they are not the same key and they will encoded to different values.

Reply:
OK, thanks for the explanation.